src/main/java/org/opentripplanner/graph_builder/module/geometry/GeometryAndBlockProcessor.java

```diff
@@ -261,13 +261,6 @@ private void interline(Collection<TripPattern> tripPatterns, Graph graph) {
             }
         }
 
-        // Copy all interline relationships into the field holding them in the graph.
-        // TODO: verify whether we need to be keeping track of patterns at all here, or could just accumulate trip-trip relationships.
-        for (P2<TripPattern> patterns : interlines.keySet()) {
-            for (P2<Trip> trips : interlines.get(patterns)) {
-                graph.interlinedTrips.put(trips.first, trips.second);
-            }
-        }
```

Comment:
Why did you remove this as part of this PR?

Reply:
This was the only field using `HashBiMap`, which for some reason did not serialize correctly with our `HashBiMapSerializer`. I thought it is easier to remove the unused field, rather than try to get the serialization working.

Reply:
Correct me if I'm wrong, but if you remove this part the whole function is useless (it just compute interline map and drop it).
Is it possible that the interline features using block id hasn't been ported yet to otp-2?

For me it seems an important one (even if probably partially supported with stay seated transfers).

The hash bi map could be replaced with an hashmap and two put statement?

Also it seems that the current implement (hashbimap) would probably fail if more than 2 consecutive trips are interlined together t1,t2,t3 with t2 inserted twice and override ref to t1.

Probably fine to only keep t1->t2 and t2->t3 reference in a regular map and query it appropriately when Fw/Bw search.

Reply:
Currently those are not read from the graph. Instead here, we should be generating [ConstrainedTransfer](https://github.com/opentripplanner/OpenTripPlanner/blob/dev-2.x/src/main/java/org/opentripplanner/model/transfer/ConstrainedTransfer.java)s. [Currently](https://github.com/opentripplanner/OpenTripPlanner/blob/d9bb509037237afa32c55acb853fee6c97123473/src/main/java/org/opentripplanner/gtfs/mapping/TransferMapper.java#L180-L186) those are created only from transfers in the transfers.txt file, but this code could be used to generate those in the future.